fdk_rdf_parser_service/app.py

```diff
@@ -1,52 +1,52 @@
 """Package for exposing validation endpoint and starting rabbit consumer."""
+from contextlib import asynccontextmanager
 import logging
+from fastapi import Body, FastAPI, HTTPException, Response, status
+import simplejson
+from fdk_rdf_parser_service.config import setup_logging
 
-from aiohttp import web
-from aiohttp_middlewares import cors_middleware, error_middleware
-
-from fdk_rdf_parser_service.config import init_logger
-from fdk_rdf_parser_service.endpoints import ping, ready
-from fdk_rdf_parser_service.kafka import avro, producer as kafka_producer
-from fdk_rdf_parser_service.rabbit import consumer as rabbit_consumer
-
-
-async def create_app(logger: logging.Logger) -> web.Application:
-    """Create a web application."""
-    logging.info("Creating web app.")
-    app = web.Application(
-        middlewares=[
-            cors_middleware(allow_all=True),
-            error_middleware(),  # default error handler for whole application
-        ],
-        logger=logger,
-    )
-
-    app.on_startup.append(rabbit_consumer.listen)
-    app.on_cleanup.append(rabbit_consumer.close)
-    app.on_startup.append(avro.setup_avro)
-    app.on_startup.append(kafka_producer.create)
-    app.on_cleanup.append(kafka_producer.shutdown)
-
-    logging.info("Setting up ping and ready endpoints.")
-    app.add_routes(
-        [
-            web.get("/ping", ping),
-            web.get("/ready", ready),
-        ]
-    )
-
-    return app
-
-
-def main() -> None:
-    """Main function for service."""
-    logger = init_logger()
-    try:
-        web.run_app(create_app(logger), print=logger.debug)
-    except Exception as e:
-        logging.error(f"Exception in main: {e}")
-        raise
+from fdk_rdf_parser_service.model import catalog_type_map
+from fdk_rdf_parser_service.service import parse_resource
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    setup_logging()
+    logging.info("Starting app")
+    yield
+
+
+app = FastAPI(
+    title="fdk-rdf-parser-service",
+    description="Services that receives RDF graphs and parses them to JSON.",
+    lifespan=lifespan,
+)
 
 
-if __name__ == "__main__":
-    main()
+@app.get("/ping")
+def get_ping():
+    """Ping route function."""
+    return Response(content="OK", status_code=200)
+
+
+@app.get("/ready")
+def get_ready():
+    """Ready route function."""
+    return Response(content="OK", status_code=200)
+
+
+@app.post("/{catalog_type}")
+def handle_request(
+    body: str = Body(..., media_type="text/turtle"), catalog_type: str | None = None
+):
+    ensured_catalog_type = catalog_type_map.get(catalog_type) if catalog_type else None
```

Comment:
Er ikke dette det samme?
```suggestion
    ensured_catalog_type = catalog_type_map.get(catalog_type)
```

Reply:
eller er ikke None gyldig som potensiell key?

Reply:
nei, den må ha str

Reply:
Hmm. I følge typesjekkingen krever den det. Men funker i terminalen med None.

Reply:
Det er nok bare python som godtar det aller meste i runtime